src/kernel/boot.c

```diff
@@ -553,6 +553,24 @@ BOOT_CODE tcb_t *create_initial_thread(cap_t root_cnode_cap, cap_t it_pd_cap, vp
     return tcb;
 }
 
+#ifndef clock_sync_test
```

Comment:
Effectively this seems to be something that can be enabled or disabled. So, why not do this in `include/kernel/boot.h`:
```
#if defined(CONFIG_DEBUG_BUILD) && defined(ENABLE_SMP_SUPPORT) && defined(CONFIG_KERNEL_MCS)
#define ENABLE_SMP_CLOCK_SYNC_TEST_ON_BOOT
#endif
```
And then it would simply be `#ifdef ENABLE_SMP_CLOCK_SYNC_TEST_ON_BOOT` everywhere. One day, this could be made a config option in CMake even, so one could also enable this for release kernels have have them halt if this fails.